Tests/YamsTests/NodeTests.swift

```diff
@@ -59,7 +59,7 @@ class NodeTests: XCTestCase {
         let scalarFloat: Node = "1.0"
         XCTAssertEqual(scalarFloat.float, 1.0)
 
-        let scalarNull: Node = "null"
+        let scalarNull = Node("null", .implicit, .plain)
         XCTAssertEqual(scalarNull.null, NSNull())
```

Comment:
I had to make this change here since the construct method will no longer construct `NSNull` if the style isn't `.plain`. 

Have I missed anything here, is that ok? I'm not familiar how this is used elsewhere but I _think_ its ok? Do we have to worry about styles like `.any`? Or is that more only for when using? 

This is the only thing that makes me wonder if the patch should have been inside `_Decoder.decodeNil()` instead as I'm not sure if it was intentional or not?

Let me know if there are better tests I can add to verify this.